### PR TITLE
chore: safe type assersions

### PR DESCRIPTION
**Pull request by OladapoAjala**
Fixes: #7672 
**Related**: _Relevant tracking issues, for context_
**Merge before/after**: _Dependent or prerequisite PRs_

**Description**

### CODE REVIEW

File: pkg/skaffold/parser/config.go [hunk 1]
```diff
@@ -134,7 +134,12 @@ func getConfigs(ctx context.Context, cfgOpts configOpts, opts config.SkaffoldOpt
 	// add profiles to record and validate that config names are unique if specified
 	seen := make(map[string]bool)
 	for _, cfg := range parsed {
-		config := cfg.(*latest.SkaffoldConfig)
+		config, ok := cfg.(*latest.SkaffoldConfig)
+		if !ok {
+			log.Entry(context.TODO()).Debugf("failed interface conversion: %v is %T not *latest.SkaffoldConfig", cfg, cfg)
+			return nil, nil, fmt.Errorf("failed interface conversion: %v is %T not *latest.SkaffoldConfig", cfg, cfg)
```
[the remark this answers is not in the document]
Comment (OladapoAjala) [reply pkg/skaffold/parser/config.go:140]: I was thinking 

`CONFIG_PARSE_ERR`

[the remark this answers is not in the document]
Comment (OladapoAjala) [reply pkg/skaffold/parser/config.go]: @tejal29 I just added an actionable error. Let me know if this suffices or I still need to create the new actionable error?

[the remark this answers is not in the document]
Comment (tejal29) [reply pkg/skaffold/parser/config.go]: We want this error code to be unique to distinguish between 
1) when skaffold.yaml is correct
2) when skaffold code failed to upgrade the config

Lets create a new one if its ok with you?

**Comment by OladapoAjala** [comment]
Hi @tejal29, I have added an actionable error. Thanks

**Review by tejal29 — APPROVED** [review]

**Comment by tejal29** [comment]
Thank you for your contribution!! Will merge if tests are green

**Comment by tejal29** [comment]
Please feel free to change this PR "Ready for Review"

File: pkg/skaffold/schema/errors/errors.go [hunk 2]
```diff
@@ -200,3 +200,19 @@ func ConfigUnknownAPIVersionErr(version string) error {
 			},
 		})
 }
+
+// SkaffoldConfigUpgradeErr specifies that skaffold config needs to be upgraded to the latest version.
+func SkaffoldConfigUpgradeErr(version string) error {
+	msg := fmt.Sprintf("skaffold config version is %s not *latest.SkaffoldConfig", version)
```
Comment (gsquared94) [inline pkg/skaffold/schema/errors/errors.go:206]: I think that the error message should say that `skaffold cannot auto-upgrade the config from version %s to version %s`. And the suggestion text should be similar to `upgrade skaffold config version to %s manually. Refer to the schema documentation at https://skaffold.dev/docs/references/yaml/`

**Review by tejal29 — APPROVED** [review]

**Pull request merged**